tests/unit/data/hf_datasets/test_preference_dataset.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import tempfile
+
+import pytest
+
+from nemo_rl.data.hf_datasets.preference_dataset import PreferenceDataset
+
+
+@pytest.fixture
+def mock_preference_data():
+    """Create temporary preference dataset files with sample data."""
+    preference_data = [
+        {
+            "context": [
+                {"role": "user", "content": "What is 2+2?"}
+            ],
+            "completions": [
+                {
+                    "rank": 1,
+                    "completion": [
+                        {"role": "assistant", "content": "The answer is 4."}
+                    ]
+                },
+                {
+                    "rank": 2,
+                    "completion": [
+                        {"role": "assistant", "content": "I don't know."}
+                    ]
+                }
+            ]
+        },
+        {
+            "context": [
+                {"role": "user", "content": "What is the capital of France?"}
+            ],
+            "completions": [
+                {
+                    "rank": 1,
+                    "completion": [
+                        {"role": "assistant", "content": "The capital of France is Paris."}
+                    ]
+                },
+                {
+                    "rank": 2,
+                    "completion": [
+                        {"role": "assistant", "content": "The capital of France is London."}
+                    ]
+                },
+            ]
+        }
+    ]
+
+    with tempfile.NamedTemporaryFile(
+        mode="w", suffix=".json", delete=False
+    ) as preference_file:
+        json.dump(preference_data, preference_file)
+        preference_path = preference_file.name
+    
+    yield preference_path
+    # Cleanup
+    os.unlink(preference_path)
```

Comment:
Should this be the following to ensure the cleanup always occur?
```suggestion
    try:
        yield preference_path
    finally:
        # Cleanup
        os.unlink(preference_path)
```